### PR TITLE
#23331 publish experiment content on start

### PR DESCRIPTION
**Pull request by dsilvam**
Includes: 

* Publishing all the content in the different variants created for the experiment
* Integration test

### CODE REVIEW

**Review by victoralfaro-dotcms — APPROVED** [review]

**Review by oidacra — APPROVED** [review]

File: dotCMS/src/main/java/com/dotcms/content/elasticsearch/business/ESContentletAPIImpl.java [hunk 1]
```diff
@@ -1176,6 +1176,20 @@ public List<Contentlet> searchByIdentifier(String luceneQuery, int limit, int of
         return contents;
     }
 
+    @Override
+    public List<Contentlet> getAllContentByVariants(final User user,
+            final boolean respectFrontendRoles, final String...variantNames) throws DotDataException, DotStateException,
+            DotSecurityException {
+
+        final String OR = Arrays.stream(variantNames).map((variant)->"variant:"+variant)
```
Comment (freddyucv) [inline dotCMS/src/main/java/com/dotcms/content/elasticsearch/business/ESContentletAPIImpl.java:1184]: I don't know if this is a good variable name

File: dotCMS/src/main/java/com/dotcms/experiments/business/ExperimentsAPIImpl.java [hunk 2]
```diff
@@ -346,9 +347,24 @@ public Experiment start(String experimentId, User user)
...
+        Experiment running = experimentToStart.withStatus(Status.RUNNING);
+        running = save(running, user);
 
+        publishContentOnExperimentVariants(user, running);
+
+        return running;
+
+    }
+
+    private void publishContentOnExperimentVariants(User user, Experiment running)
```
Comment (freddyucv) [inline dotCMS/src/main/java/com/dotcms/experiments/business/ExperimentsAPIImpl.java:359]: running is not a good name

File: dotCMS/src/main/java/com/dotmarketing/portlets/contentlet/business/ContentletAPI.java [hunk 3]
```diff
@@ -447,6 +447,10 @@ public Contentlet copyContentlet(Contentlet contentlet, Host host, User user, bo
 	 */
 	public List<Contentlet> search(String luceneQuery, int limit, int offset, String sortBy, User user, boolean respectFrontendRoles, int requiredPermission) throws DotDataException, DotSecurityException;
 
+	List<Contentlet> getAllContentByVariants(final User user,
```
Comment (freddyucv) [inline dotCMS/src/main/java/com/dotmarketing/portlets/contentlet/business/ContentletAPI.java:450]: doc

**Review by freddyucv — APPROVED** [review]

File: dotCMS/src/main/java/com/dotcms/experiments/business/ExperimentsAPIImpl.java (hunk 2 above)
Comment (dsilvam) [reply dotCMS/src/main/java/com/dotcms/experiments/business/ExperimentsAPIImpl.java:359]: Done

File: dotCMS/src/main/java/com/dotmarketing/portlets/contentlet/business/ContentletAPI.java (hunk 3 above)
Comment (dsilvam) [reply dotCMS/src/main/java/com/dotmarketing/portlets/contentlet/business/ContentletAPI.java:450]: Done

File: dotCMS/src/main/java/com/dotcms/content/elasticsearch/business/ESContentletAPIImpl.java (hunk 1 above)
Comment (dsilvam) [reply dotCMS/src/main/java/com/dotcms/content/elasticsearch/business/ESContentletAPIImpl.java]: Done

**Pull request merged**